### ISSUE TITLE
TMC reports failing to load at St Clair and Oriole

### ISSUE THREAD

**Issue by candu** [CityofToronto/bdit_flashcrow#682]
**What Happened**
When attempting to load the TMC Summary Report at [St Clair and Oriole](https://move.intra.dev-toronto.ca/view/location/s1:AahsmB/POINTS/reports/TMC), the report does not load, and the loading spinner spins indefinitely.

**What Should Happen**
The report should load successfully.

**To Reproduce**
Steps to reproduce the bug:
1. Go to [TMC reports at St Clair and Oriole](https://move.intra.dev-toronto.ca/view/location/s1:AahsmB/POINTS/reports/TMC).

At this point, the report does not load, and devtools report HTTP 500 from the `reporter` backend.

**Additional Notes**
All other TMC report types seem to work fine here; only the summary report fails.

Stacktrace from `reporter` in DEV0:

```
Debug: internal, implementation, error
    TypeError: Cannot destructure property 'data' of 'undefined' as it is undefined.
    at /home/ec2-user/flashcrow/lib/reports/ReportCountSummaryTurningMovement.js:22:54
    at Array.map (<anonymous>)
    at Function.sumIndices (/home/ec2-user/flashcrow/lib/reports/ReportCountSummaryTurningMovement.js:22:47)
    at Function.sumSection (/home/ec2-user/flashcrow/lib/reports/ReportCountSummaryTurningMovement.js:63:51)
    at Function.transformCountData (/home/ec2-user/flashcrow/lib/reports/ReportCountSummaryTurningMovement.js:143:50)
    at ReportCountSummaryTurningMovement.transformData (/home/ec2-user/flashcrow/lib/reports/ReportCountSummaryTurningMovement.js:187:53)
    at ReportCountSummaryTurningMovement.generate (/home/ec2-user/flashcrow/lib/reports/ReportBase.js:126:34)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async handler (/home/ec2-user/flashcrow/lib/controller/ReportController.js:120:26)
```

**Comment by candu** [comment]
Most TMCs in FLOW have 32 data rows: one for every 15-minute block of an 8-hour count.  This location only has 31 data rows:

![St Clair and Oriole only has 31 data rows, not 32](https://user-images.githubusercontent.com/151605/98726688-d7d9a980-2364-11eb-8f14-3fa0696ec132.png)

However, `ReportCountSummaryTurningMovement` hard-codes the indices for AM / PM peaks, and `ArrayUtils.selectIndices` doesn't check for out-of-bounds conditions.

We should probably do two things here:

- enforce array index bounds in `ArrayUtils.selectIndices`, which is unique among `ArrayUtils` methods in accessing array elements by index;
- update `ReportCountSummaryTurningMovement` to auto-detect AM / PM peaks (e.g. as greatest consecutive 2-hour total volume).

**Comment by candu** [comment]
Can confirm that this bug still exists, but fixing it is more work than we have time for pre-1.2.  This should be fixed by addressing #837 .

**Comment by candu** [comment]
Fixed by addressing #837 , as anticipated.
